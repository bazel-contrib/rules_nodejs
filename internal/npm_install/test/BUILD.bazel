load("@build_bazel_rules_nodejs//:defs.bzl", "jasmine_node_test", "nodejs_binary")

filegroup(
    name = "node_modules",
    srcs = glob(["node_modules/**"]),
)

filegroup(
    name = "goldens",
    srcs = glob(["golden/**"]),
)

jasmine_node_test(
    name = "test",
    srcs = glob(["*.spec.js"]),
    data = [
        ":check.js",
        ":goldens",
        ":node_modules",
        ":yarn.lock",
    ],
    node_modules = ":node_modules",
    deps = ["//internal/npm_install:generate_build_file"],
)

nodejs_binary(
    name = "test.accept",
    data = [
        ":check.js",
        ":goldens",
        ":node_modules",
        ":update_golden.js",
        ":yarn.lock",
        "//internal/npm_install:generate_build_file",
    ],
    entry_point = "build_bazel_rules_nodejs/internal/npm_install/test/update_golden.js",
    install_source_map_support = False,
    node_modules = ":node_modules",
)
