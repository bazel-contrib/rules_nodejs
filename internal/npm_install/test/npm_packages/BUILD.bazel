load("@npm//bazel_workspaces_consistent:index.bzl", "some_rule")

some_rule(name = "test_data")
