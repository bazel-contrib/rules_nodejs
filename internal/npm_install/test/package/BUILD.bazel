# Generated file from yarn_install/npm_install rule.
# See rules_nodejs/internal/npm_install/generate_build_file.ts

load("@build_bazel_rules_nodejs//:index.bzl", "js_library")

# All rules in other repositories can use these targets
package(default_visibility = ["//visibility:public"])

# The node_modules directory in one catch-all js_library.
# NB: Using this target may have bad performance implications if
# there are many files in target.
# See https://github.com/bazelbuild/bazel/issues/5153.
js_library(
    name = "node_modules",
)
