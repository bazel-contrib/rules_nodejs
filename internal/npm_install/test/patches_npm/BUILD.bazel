load("//:index.bzl", "nodejs_test")

nodejs_test(
    name = "test",
    data = [
        "@internal_npm_install_test_patches_npm//:_/internal/npm_install/test/patches_npm/package.json",
        "@internal_npm_install_test_patches_npm//semver",
    ],
    entry_point = "test.js",
)
