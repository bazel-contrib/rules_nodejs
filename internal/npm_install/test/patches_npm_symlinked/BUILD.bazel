load("//:index.bzl", "nodejs_test")

nodejs_test(
    name = "test",
    data = [
        "@internal_npm_install_test_patches_npm_symlinked//semver",
    ],
    entry_point = "test.js",
)
