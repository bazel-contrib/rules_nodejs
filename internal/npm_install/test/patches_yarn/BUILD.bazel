load("//:index.bzl", "nodejs_test")

nodejs_test(
    name = "test",
    data = [
        "@internal_npm_install_test_patches_yarn//:_/internal/npm_install/test/patches_yarn/package.json",
        "@internal_npm_install_test_patches_yarn//semver",
    ],
    entry_point = "test.js",
)
