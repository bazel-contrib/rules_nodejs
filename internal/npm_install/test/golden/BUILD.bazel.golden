
package(default_visibility = ["//visibility:public"])
filegroup(
    name = "node_modules",
    srcs = glob(["node_modules/.bin/*"]) + [
        "//node_modules/balanced-match:balanced-match__lite",
        "//node_modules/brace-expansion:brace-expansion__lite",
        "//node_modules/concat-map:concat-map__lite",
        "//node_modules/diff:diff__lite",
        "//node_modules/fs.realpath:fs.realpath__lite",
        "//node_modules/glob:glob__lite",
        "//node_modules/inflight:inflight__lite",
        "//node_modules/inherits:inherits__lite",
        "//node_modules/jasmine:jasmine__lite",
        "//node_modules/jasmine-core:jasmine-core__lite",
        "//node_modules/minimatch:minimatch__lite",
        "//node_modules/once:once__lite",
        "//node_modules/path-is-absolute:path-is-absolute__lite",
        "//node_modules/unidiff:unidiff__lite",
        "//node_modules/wrappy:wrappy__lite",
        "//node_modules/@gregmagolan/test-a:test-a__lite",
        "//node_modules/@gregmagolan/test-b:test-b__lite",
        "//node_modules/@yarnpkg/lockfile:lockfile__lite",
    ],
)
